# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/ExternalEngineStateMachine.cpp
# Commit: 35d86243cb45
# Full Hash: 35d86243cb45f7e02dd6ec0eea04cd62c2a70196
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2024-09-16 09:14:26
# Regressor Bug: 1908245
# File Overlap Count: 1
# Description:
#   Bug 1908245 - Add support for missing requestVideoFrameCallback parameters. r=media-playback-reviewers,webidl,padenot,saschanaz
#   
#   This patch adds support for exposing the capture time, receive time,
#   processing duration and RTP timestamp parameters on the callback for
#   rVFC for WebRTC related elements.
# ==============================================================================

diff -r 7d143d92bf2a -r 35d86243cb45 dom/media/ExternalEngineStateMachine.cpp
--- a/dom/media/ExternalEngineStateMachine.cpp	Sun Sep 15 20:03:40 2024 +0000
+++ b/dom/media/ExternalEngineStateMachine.cpp	Sun Sep 15 21:49:38 2024 +0000
@@ -977,10 +977,12 @@
             // Send image to PIP window.
             if (mSecondaryVideoContainer.Ref()) {
               mSecondaryVideoContainer.Ref()->SetCurrentFrame(
-                  mVideoDisplay, aVideo->mImage, TimeStamp::Now());
+                  mVideoDisplay, aVideo->mImage, TimeStamp::Now(),
+                  media::TimeUnit::Invalid(), aVideo->mTime);
             } else {
               mVideoFrameContainer->SetCurrentFrame(
-                  mVideoDisplay, aVideo->mImage, TimeStamp::Now());
+                  mVideoDisplay, aVideo->mImage, TimeStamp::Now(),
+                  media::TimeUnit::Invalid(), aVideo->mTime);
             }
           },
           [this, self](const MediaResult& aError) {